webapp/src/Utils/Scoreboard/SingleTeamScoreboard.php

```diff
@@ -100,7 +100,7 @@ protected function calculateScoreboard()
 
             $this->matrix[$scoreRow->getTeam()->getTeamid()][$scoreRow->getProblem()->getProbid()] = new ScoreboardMatrixItem(
                 $scoreRow->getIsCorrect($this->restricted),
-                $scoreRow->getIsFirstToSolve(),
+                $scoreRow->getIsCorrect($this->showFts) ? $scoreRow->getIsFirstToSolve() : false,
```

Comment:
do you really want getIsCorrect here?

Reply:
I believe so. If it is not correct, we never want to show FTS (obviously). If it is correct, we want to show FTS if:

* It was before freeze
* It was during freeze but now is after unfreeeze for teams
* You are on the jury pages

The first and only parameter to `getIsCorrect` is whether we are restricted. For jury pages we set `showFts` to `true`, which corresponds to "not restricted". For team pages we set it to true only if we want to show frozen.

Or am I missing something?

Reply:
Can't it be simpler written as something along those lines:
```
$isCorrect = $scoreRow->getIsCorrect($this->restricted);
$this->matrix[$scoreRow->getTeam()->getTeamid()][$scoreRow->getProblem()->getProbid()] = new ScoreboardMatrixItem(
                $isCorrect,
                $isCorrect && $scoreRow->getIsFirstToSolve() && $this->showFts,
...
```

Reply:
That logic is incorrect. It will hide *all* first to solves, also those that appear before the freeze. I do want to show those....

Reply:
Half-related: I think it would be more natural to directly pass the `$showFts` argument to `$scoreRow->getIsFirstToSolve()`.

Reply:
I like that